tla-types/src/main/scala/at/forsyte/apalache/tla/typecheck/TypeCheckerTool.scala

```diff
@@ -0,0 +1,28 @@
+package at.forsyte.apalache.tla.typecheck
+
+import at.forsyte.apalache.tla.lir.{TlaModule, UID}
+import at.forsyte.apalache.tla.typecheck.etc._
+
+/**
+  * The API to the type checker. It first translates a TLA+ module into EtcExpr and then does the type checking.
+  *
+  * @author Igor Konnov
+  */
+class TypeCheckerTool {
+  def check(listener: TypeCheckerListener, module: TlaModule): Boolean = {
+    // TODO: remember to type check the assumptions!
```

Comment:
@konnov Is this something we should implement as part of this PR, or an additional feature we want to add later?

Reply:
This is an additional feature. We have not been paying attention to `ASSUME(...)` so far, but we have to do it at some point.

Reply:
Opened an issue to track this #330